docs/types/literal-suffixes.rst

```diff
@@ -0,0 +1,243 @@
++-------------------------------------------------------------+----------------------------------------------------------------+
+| ``address``                                                 | - :ref:`Address literals<address_literals>`                    |
++-------------------------------------------------------------+----------------------------------------------------------------+
+| ``bytes1``, ..., ``bytes32``                                | - Hexadecimal number literals (not for ``bytes20``)            |
+|                                                             | - :ref:`Hexadecimal string literals<hexadecimal_literals>`     |
+|                                                             | - :ref:`String literals<string_literals>`                      |
+|                                                             | - :ref:`Unicode literals<unicode_literals>`                    |
+|                                                             | - :ref:`Zero<rational_literals>`                               |
++-------------------------------------------------------------+----------------------------------------------------------------+
+| ``bytes``                                                   | - :ref:`Hexadecimal string literals<hexadecimal_literals>`     |
+|                                                             | - :ref:`String literals<string_literals>`                      |
+|                                                             | - :ref:`Unicode literals<unicode_literals>`                    |
++-------------------------------------------------------------+----------------------------------------------------------------+
+| ``string``                                                  | - :ref:`String literals<string_literals>`                      |
+|                                                             | - :ref:`Unicode literals<unicode_literals>`                    |
++-------------------------------------------------------------+----------------------------------------------------------------+
+
+For two-parameter suffix functions the first parameter (representing the mantissa) can be of any integer type.
+The second parameter (the exponent) must be of an unsigned integer type.
+
+.. note::
+    40-digit literals prefixed with ``0x`` such as, for example, ``0xdCad3a6d3569DF655070DEd06cb7A1b2Ccd1D3AF``
+    always represent ``address`` literals in the language.
+    To invoke a suffix accepting ``bytes20`` you must use one of the other literal kinds implicitly
+    convertible to ``bytes20``, e.g. a hexadecimal string literal
+    (``hex"dCad3a6d3569DF655070DEd06cb7A1b2Ccd1D3AF"``).
+
+.. note::
+    Suffix functions accepting ``address payable`` are not allowed since address literals are never payable.
+
+Suffix functions may not accept or return reference types with ``storage`` or ``calldata`` locations.
```

Comment:
```suggestion
For two-parameter suffix functions, the first parameter (representing the mantissa) can be of any integer type.
The second parameter (the exponent) must be of an unsigned integer type.

.. note::
    40-digit literals prefixed with ``0x`` such as, for example, ``0xdCad3a6d3569DF655070DEd06cb7A1b2Ccd1D3AF``
    always represent ``address`` literals in the language.
    To invoke a suffix accepting ``bytes20`` you must use one of the other literal kinds implicitly
    convertible to ``bytes20``, e.g. a hexadecimal string literal
    (``hex"dCad3a6d3569DF655070DEd06cb7A1b2Ccd1D3AF"``).

.. note::
    Suffix functions accepting ``address payable`` are not allowed since address literals are never payable.

Suffix functions may not accept or return reference types with ``storage`` or ``calldata`` locations.
```

Reply:
Just a general request: would it be possible to only highlight the part that actually has changes in it? As is I have to carefully scan though the whole diff I see in the comment to make sure I don't miss any change. And I've seen multiple cases where github does not properly highlight something in the diff so they're easy to miss in cases like this where the only change something small like a comma.

Reply:
I believe what you're asking is to add suggestion on a per line basis instead of per paragraph/multiline.

Reply:
I've never had that happen to me! But wow yeah, if that's the case, your request makes sense.

Reply:
That would be best for me, but I also don't want to be that picky about it. The ones that bother me are just the ones where the change seems to span many lines but the only real change is a single comma :) Then I'd prefer to have just that one line with the comma highlighted. Otherwise I'm second-guessing myself if I maybe the change is there but I just missed it :)